frontend/src/metabase/dashboard/components/DashboardGrid.jsx

```diff
@@ -113,12 +115,29 @@ class DashboardGrid extends Component {
       selectedTabId,
     );
 
+    if (!isEditing) {
+      this.setState({
+        initLayouts: this.getInitialCardLayout(cards),
+      });
+    }
+
     this.setState({
       visibleCardIds,
       layouts: this.getLayouts(cards),
     });
   }
 
+  getInitialCardLayout = cards => {
```

Comment:
nit: Maybe call this `initialCardSizes` since there are only width and height? To not mix this with `layout` term used for other cases